### PR TITLE
Api call to get interaction from desk.com

### PR DESCRIPTION
**Pull request by frj**
This PR implements the initial story for the desk.com interactions export tool, and contains the following:
- Creating the build etc for the project
- Skeleton services for the export tool
- Command line parseing
- desk.com api client that implements the call to list interactions endpoint
- Running the tool does a single call to the interactions endpoint and logs the ids of the interactions to the console to demonstrate that it works.

### CODE REVIEW

File: application/src/main/scala/com/gu/deskcomexporttool/DeskComClient.scala [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    override def getAllInteractions(page: Int, pageSize: Int): EitherT[Future, DeskComApiError, List[Interaction]] = {
+      for {
+        httpResponse <- httpClient.request(
+          HttpRequest(
+            "GET",
+            s"${config.baseUrl}/api/v2/interactions?page=$page&per_page=$pageSize&sort_field=created_at&sort_direction=asc",
+            List(authHeader)
+          )
+        ).leftMap(httpError => DeskComApiError(s"Request for interactions failed: $httpError"))
+        _ <- EitherT.fromEither(validateStatusCode(httpResponse.statusCode))
```
Comment (arausuy) [inline application/src/main/scala/com/gu/deskcomexporttool/DeskComClient.scala:43]: Do we need this?

File: application/src/main/scala/com/gu/deskcomexporttool/DeskComClient.scala [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        parsedResponseBody <- EitherT.fromEither(parseGetAllInteractions(httpResponse.body))
+      } yield parsedResponseBody._embedded.entries
+    }
+
+    private def parseGetAllInteractions(body: String): Either[DeskComApiError, GetInteractionsResponse] = {
+      decode[GetInteractionsResponse](body)
+        .left
+        .map { parsingFailure =>
```
Comment (arausuy) [inline application/src/main/scala/com/gu/deskcomexporttool/DeskComClient.scala]: I'm guessing there is no leftMap equivalent?

**Review by arausuy — APPROVED** [review]

**Comment by arausuy** [comment]
LGTM

File: application/src/main/scala/com/gu/deskcomexporttool/DeskComClient.scala (hunk 1 above)
Comment (frj) [reply application/src/main/scala/com/gu/deskcomexporttool/DeskComClient.scala:43]: validateStatusCode will cause a failure if it returns a left, if it returns a right we just ignore it and use the response from the previous step

File: application/src/main/scala/com/gu/deskcomexporttool/DeskComClient.scala (hunk 2 above)
Comment (frj) [reply application/src/main/scala/com/gu/deskcomexporttool/DeskComClient.scala]: 👍

**Pull request merged**